src/interface/guide/components/BulletGraph.tsx

```diff
@@ -0,0 +1,203 @@
+  /** The width of this performance zone as a percentage (0-100) */
+  width: number;
+  /** Background color for this zone */
+  color: string;
+  /** Optional label for this zone (e.g., "Poor", "OK", "Good") */
+  label?: string;
+}
+
+export interface BulletGraphProps {
+  /** The actual value to display */
+  actual: number;
+  /** The maximum/target value (used to calculate percentages) */
+  maximum: number;
+  /** Label to display on the main bar (e.g., "15 casts") */
+  actualLabel: string;
+  /** Label to display at the target marker (e.g., "Max 20") */
+  maximumLabel: string;
+  /** Color for the main progress bar */
+  barColor: string;
+  /** Performance ranges to display in the background */
+  performanceRanges?: PerformanceRange[];
+  /** Optional secondary metric (e.g., wasted time) */
+  secondaryMetric?: {
+    /** Value as percentage of fight duration (0-100) */
+    value: number;
+    /** Label to display (e.g., "5s capped") */
+    label: string;
+    /** Color for the secondary bar */
+    color: string;
+  };
+  /** Optional CSS class name */
+  className?: string;
+}
+
+/**
+ * BulletGraph component for visualizing performance metrics.
+ *
+ * @param actual - The actual value to display
+ * @param maximum - The maximum/target value (used for percentage calculation)
+ * @param actualLabel - Label to display on the main bar (e.g., "15 casts")
+ * @param maximumLabel - Label to display at the target marker (e.g., "Max 20")
+ * @param barColor - Color for the main progress bar
+ * @param performanceRanges - Optional performance ranges for the background
+ * @param secondaryMetric - Optional secondary metric (e.g., wasted time)
+ * @param className - Optional CSS class name
+ */
+export default function BulletGraph({
+  actual,
+  maximum,
+  actualLabel,
+  maximumLabel,
+  barColor,
+  performanceRanges,
+  secondaryMetric,
+  className,
+}: BulletGraphProps) {
+  const percentage = Math.min((actual / maximum) * 100, 100);
+
+  return (
+    <Container className={className}>
+      <GraphBackground>
+        {performanceRanges?.map((range, index) => (
+          <PerformanceZone key={index} width={range.width} color={range.color} />
+        ))}
+      </GraphBackground>
+
+      <MainBar width={percentage} color={barColor}>
+        <BarLabel>{actualLabel}</BarLabel>
+      </MainBar>
+
+      <TargetMarker>
+        <TargetLine />
+        <TargetLabel>{maximumLabel}</TargetLabel>
+      </TargetMarker>
+
+      {secondaryMetric && (
+        <SecondaryBar>
+          <SecondaryFill width={secondaryMetric.value} color={secondaryMetric.color} />
+          <SecondaryLabel position={secondaryMetric.value}>{secondaryMetric.label}</SecondaryLabel>
+        </SecondaryBar>
+      )}
+    </Container>
+  );
+}
+
+const Container = styled.div`
+  width: 100%;
+  position: relative;
+  display: flex;
+  flex-direction: column;
+  gap: 4px;
+`;
+
+const GraphBackground = styled.div`
+  width: 100%;
+  height: 32px;
```

Comment:
avoid using fixed heights for things like this. it means that the resizing the parent element doesn't actually re-size the contents and results in visual breakage.